src/esbmc/esbmc_parseoptions.cpp

```diff
@@ -1584,6 +1590,16 @@ bool esbmc_parseoptionst::process_goto_program(
       goto_k_induction(goto_functions, msg);
     }
 
+    if(cmdline.isset("goto-contractor"))
+    {
+#ifdef ENABLE_GOTO_CONTRACTOR
+      goto_contractort(goto_functions, msg);
```

Comment:
Well, you have done it here :)

What happens here is a call to the constructor `goto_contractort(...)`. This call creates an "anonymous" and in this case also "temporary" object, because it not assigned to anything. Here it is equivalent to code that would create a named object like so: `goto_contractort var(...)` and there is no difference between the two, because the the lifetime of this `var` object directly ends on the next line as does that of the temporary version.

So it runs constructor followed by destructor, ok, fine; but it requires a bit more thinking (at least for me) to understand what's going on. My suggestion was to wrap this into a stand-alone function and call that, maybe like so:
```c++
void goto_contractor(...)
{
#ifdef ENABLE_GOTO_CONTRACTOR
  goto_contractort(...); /* all functionality invoked by constructor */
#else
  msg.error(...);
  abort();
#endif
}
```
It's just a suggestion; it might already be enough to just add such a comment in order to minimize confusion. The advantage of having such a function is that you could also move the invocation of the algorithm (labelled by messages "1/5", "2/5", etc.) from the constructor to this function and would have a more object-oriented approach in that external algorithms could just work with an object of type `goto_contractort` instead of embedding the algorithm inside it.

Reply:
Thanks for the feedback. I have changed it now to work like other options in ESBMC. Particularly, similar to k-induction where there's a function call called goto_contractor and inside it will crate the object goto_contractort. Also added the #ifdef guards.